packages/primitives/tooltip/src/tooltip-root.directive.ts

```diff
@@ -0,0 +1,285 @@
+            return;
+        }
+
+        if (this.isOpenDelayed()) {
+            this.handleDelayedOpen();
+        } else {
+            this.handleOpen();
+        }
+    }
+
+    onTriggerLeave(): void {
+        window.clearTimeout(this.openTimer);
+        this.handleClose();
+    }
+
+    onOpen(): void {
+        window.clearTimeout(this.skipDelayTimer);
+        this.isOpenDelayed.set(false);
+    }
+
+    onClose(): void {
+        window.clearTimeout(this.skipDelayTimer);
+
+        this.skipDelayTimer = window.setTimeout(() => {
+            this.isOpenDelayed.set(true);
+        }, this.tooltipConfig.skipDelayDuration);
+    }
+
+    handleOpen(): void {
+        if (this.isControlledExternally) {
+            return;
+        }
+
+        this.wasOpenDelayed.set(false);
+        this.setOpen(true);
+    }
+
+    handleClose(): void {
+        if (this.isControlledExternally) {
+            return;
+        }
+
+        window.clearTimeout(this.openTimer);
+        this.setOpen(false);
+    }
+
+    private handleOverlayKeydown(): void {
+        if (!this.overlayRef) {
+            return;
+        }
+
+        this.overlayRef
+            .keydownEvents()
+            .pipe(
+                filter((event) => event.key === 'Escape'),
+                takeUntilDestroyed(this.destroyRef)
+            )
+            .subscribe((event) => {
+                this.tooltipContentDirective().onEscapeKeyDown.emit(event);
+
+                if (!event.defaultPrevented) {
+                    this.handleClose();
+                }
+            });
+    }
+
+    private handlePointerDownOutside(): void {
+        if (!this.overlayRef) {
+            return;
+        }
+
+        this.overlayRef
+            .outsidePointerEvents()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((event) => this.tooltipContentDirective().onPointerDownOutside.emit(event));
+    }
+
+    private handleDelayedOpen(): void {
+        window.clearTimeout(this.openTimer);
+
+        this.openTimer = window.setTimeout(() => {
+            this.wasOpenDelayed.set(true);
+            this.setOpen(true);
+        }, this.delayDuration());
+    }
+
+    private setOpen(open = false): void {
+        if (open) {
+            this.onOpen();
+
+            document.dispatchEvent(new CustomEvent('tooltip.open'));
+        } else {
+            this.onClose();
+        }
+
+        this.isOpen.set(open);
+        this.onOpenChange.emit(open);
+    }
+
+    private createOverlayRef(): OverlayRef {
+        if (this.overlayRef) {
+            return this.overlayRef;
+        }
+
+        this.overlayRef = this.overlay.create({
+            direction: undefined,
+            positionStrategy: this.getPositionStrategy(this.tooltipContentDirective().position()),
+            scrollStrategy: this.overlay.scrollStrategies.close()
+        });
+
+        this.overlayRef
+            .detachments()
+            .pipe(takeUntilDestroyed(this.destroyRef))
```

Comment:
it seems that it is enough to get only one emit
```suggestion
            .pipe(take(1))
```

Reply:
using both for now. `take(1)` most likely will be triggered sooner then `takeUntilDestryed`. but i can't guarantee that streem emit value